test/index.js

```diff
@@ -1463,6 +1463,139 @@ describe('Nightmare', function () {
       didSubscribe.should.be.false;
     });
 
+    it('should subscribe to frames when requested necessary', function(done) {
+      var didSubscribe = false;
+      var didUnsubscribe = false;
+      var FrameManager = require('../lib/frame-manager.js');
+      var fn;
+      var manager = FrameManager({
+        webContents: {
+          debugger: {
+            isAttached: function() { return true; },
+            sendCommand: function(command) { if (command === 'DOM.highlightRect') { fn('mock-data'); }}
+          },
+          beginFrameSubscription: function(_fn) { didSubscribe = true; fn = _fn; },
+          endFrameSubscription: function() { didUnsubscribe = true; },
+          executeJavaScript: function() {}
+        }
+      });
```

Comment:
Maybe we could break out the frame manager tests into their own file altogether? It's only 1 test prior this this, it would eliminate worries, and it would more towards a cleaner repo (i.e. 1 test file per 1 repo file)